### ISSUE TITLE
Why does my Json Deserializer return 0 for all numeric values, but Swagger returns the correct values?

### ISSUE THREAD

**Issue by rfalanga** [dotnet/docs#43729]
### Type of issue

Other (describe below)

### Description

I'm working on writing a Blazor app using .NET 8. I've got a database with column names that have embedded whitespace. When I deserialize it all the numeric values (SMALLINT in a SQL Server database) are 0 (for example the column "Video Number"), which is wrong and doesn't correctly show what's in the table. However, when I check the values using Swagger, it is perfectly correct. I had thought that the attribute JsonPropertyName would resolve my issue, but it doesn't. Here's my class 

```csharp
public partial class Entry
{
    [Key, Column("Video Number", Order = 0)]
    [JsonPropertyName("Video Number")]
    public short VideoNumber { get; set; }  

    [Key, Column("Entry", Order = 1)]
    public byte Entry1 { get; set; }

    [Required]
    public short ShowCode { get; set; }

    [Required]
    public string EpisodeName { get; set; } = null!;

    public DateTime? ActualPlayTime { get; set; }

    public bool Completed { get; set; }

    public string? Comment { get; set; }

    public TimeOnly? ActualPlayTime1 { get; set; }

    public virtual Show Show { get; set; } = null!;

    public virtual Video VideoNumberNavigation { get; set; } = null!;
}
```

And here's the code snippet where I'm trying to read the data out of the table:

```csharp
var contentStream = await httpResponse.Content.ReadAsStreamAsync();
var bozo = await System.Text.Json.JsonSerializer.DeserializeAsync<Entry[]>(contentStream, new System.Text.Json.JsonSerializerOptions { DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull, PropertyNameCaseInsensitive = true });
return bozo;
```

### Page URL

https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/customize-properties

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/standard/serialization/system-text-json/customize-properties.md

### Document Version Independent Id

27c18218-d28f-a524-b7e1-8262005b7026

### Article author

@gewarren

### Metadata

* ID: 7bee4ecb-7464-b321-937a-e6d4586556eb 
* Service: **dotnet-fundamentals**

[Related Issues](https://github.com/dotnet/docs/issues?q=is%3Aissue+is%3Aopen+27c18218-d28f-a524-b7e1-8262005b7026)

**Comment by gewarren** [comment]
@eiriktsarpalis Do you know what's going on here?

**Comment by eiriktsarpalis** [comment]
If I'm understaning the issue correctly, this is not related to the documentation per se but is a product question. @rfalanga Would it be possible to share the actual JSON you're trying to deserialize, how you're generating swagger files (Swashbuckle? Asp.NET OpenAPI? Something else?) and the expected vs actual results? Thanks.

**Comment by rfalanga** [comment]
I'm sorry, I'd forgotten that I posted this question here. Thank you. @eiriktsarpalis, for the reminder.

I've discovered the problem. I had mistakenly tried to retrieve values from the wrong table. And because of that, the numeric values were all 0, since none of those columns were in the table I was trying to pull data from. However, coincidentally, the string values were coming from the table I was pulling data from, thus my confusion.

Bottom line, it was my fault, and I apologize for bringing this up. I'll close this issue now.
